actions/checkForAndPlaceOrder.ts

```diff
@@ -510,22 +524,43 @@ async function _placeOrder(
   return { result: PollResultCode.SUCCESS };
 }
 
-function _handleOrderBookError(
-  status: any,
-  data: any,
-  error: any
-):
-  | Omit<PollResultSuccess, "order" | "signature">
-  | PollResultTryNextBlock
-  | PollResultUnexpectedError {
+function _handleOrderBookError(params: {
+  status: any;
+  data: any;
+  error: any;
+  blockTimestamp: number;
+}): Omit<PollResultSuccess, "order" | "signature"> | PollResultErrors {
+  const { status, data, error, blockTimestamp } = params;
   if (status === 400) {
     // The order is in the OrderBook, all good :)
-    if (data?.errorType === "DuplicatedOrder") {
+    if (data?.errorType === ApiErrors.DUPLICATE_ORDER) {
       return {
         result: PollResultCode.SUCCESS,
       };
     }
 
+    // It's possible that an order has not enough allowance or balance.
+    // Returning DONT_TRY_AGAIN would be to drastic, but we can give the WatchTower a break by scheduling next attempt in a few minutes
+    // This why, we don't so it doesn't try in every block
+    if (
+      [
+        ApiErrors.INSUFFICIENT_ALLOWANCE,
+        ApiErrors.INSUFFICIENT_BALANCE,
+      ].includes(data?.errorType)
+    ) {
+      const nextPollTimestamp =
+        blockTimestamp + WAITING_TIME_MINUTES_FOR_NOT_BALANCE * 60;
+      return {
+        result: PollResultCode.TRY_AT_EPOCH,
+        epoch: blockTimestamp + WAITING_TIME_MINUTES_FOR_NOT_BALANCE,
```

Comment:
Wait, this is not in seconds.
Why not use `nextPollTimestamp` here?

Reply:
Also, maybe the const should be in seconds already to avoid this kind of issue?

Reply:
Not good reason, just call it i was dumb